WalletWasabi/Wallets/P2pBlockProvider.cs

```diff
@@ -74,12 +76,14 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 				cancellationToken.ThrowIfCancellationRequested();
 				try
 				{
-					// Try to get block information from local running Core node first.
-					block = await TryDownloadBlockFromLocalNodeAsync(hash, cancellationToken).ConfigureAwait(false);
+					if(!LastFailConnectLocalNode.IsRunning || LastFailConnectLocalNode.Elapsed.TotalSeconds >= RuntimeParams.Instance.RetryConnectLocalNodeSeconds) { 
```

Comment:
> `GetBlockAsync` method seems like it can be called multiple times (why not?). Or am I missing something?

The point is to wait for N seconds before retrying, even during a new call. 
But you brought a good point, I have to add a lock otherwise the logs will look bad in cases of concurrent calls